java/src/main/native/src/jni_compiled_expr.hpp

```diff
@@ -38,8 +38,8 @@ class compiled_expr {
   /** All column reference nodes within the expression tree */
   std::vector<std::unique_ptr<cudf::ast::column_reference>> column_refs;
 
-  /** All expression nodes within the expression tree */
-  std::vector<std::unique_ptr<cudf::ast::expression>> expressions;
+  /** All operation nodes within the expression tree */
+  std::vector<std::unique_ptr<cudf::ast::operation>> operations;
```

Comment:
Now that the base expression class has a virtual destructor, there's no need to track the different node types separately.  These can all be in a single `std::vector<std::unique_ptr<cudf::ast::expression>>`.  We still need to track the scalars corresponding to AST literals separately, but otherwise everything else can be combined.  I can do this in a followup PR if you'd rather not refactor that here.